app/services/rbac_api.rb

```diff
@@ -1,32 +1,63 @@
 # frozen_string_literal: true
 
-# This class is meant to handle calls to the RBAC light API
+require 'insights-rbac-api-client'
+
+# This class is meant to handle calls to the RBAC API
 class RbacApi
+  COMPLIANCE_FULL_ACCESS = 'compliance:*:*'
+  APPS = 'compliance,inventory'
+  ANY = '*'
+
   def initialize(b64_identity)
-    @url = URI.parse("#{URI.parse(Settings.rbac_url)}"\
-                     "#{Settings.path_prefix}/rbac/v1/access/")
+    @client = RBACApiClient::AccessApi.new
     @b64_identity = b64_identity
   end
 
-  def check_user
+  # rubocop:disable Metrics/MethodLength
+  def load_users_permissions
+    opts = {
+      limit: 10,
+      offset: 0
+    }
     begin
-      body = JSON.parse(access_check_response.body)
-      body['data'].find do |access_check|
-        return true if access_check['permission'] == 'compliance:*:*'
-      end
-    rescue Faraday::ClientError => e
-      Rails.logger.info("#{e.message} #{e.response}")
+      result = @client.get_principal_access(
+        RbacApi::APPS,
+        opts.merge({ header_params: { 'X_RH_IDENTITY': @b64_identity } })
+      )
+      result.data
+    rescue RBACApiClient::ApiError => e
+      Rails.logger.info(e.message)
+      # TODO: unauthorized
```

Comment:
This HAS to be covered, we can't merge it without. Our API should return a 403 if the RBAC service is unavailable.

Reply:
Maybe you could mock the result somehow for future consumption?